src/Std/Data/DTreeMap/Internal/Lemmas.lean

```diff
@@ -4353,6 +4370,99 @@ theorem minKey?_le_minKey?_erase! [TransOrd α] (h : t.WF) {k km kme} :
     compare km kme |>.isLE := by
   simpa only [erase_eq_erase!] using minKey?_le_minKey?_erase h
 
+theorem minKey?_insertIfNew [TransOrd α] (h : t.WF) {k v} :
+    (t.insertIfNew k v h.balanced).impl.minKey? =
+      t.minKey?.elim k fun k' => if compare k k' = .lt then k else k' := by
+  simp_to_model [insertIfNew] using List.minKey?_insertEntryIfNew
+
+theorem minKey?_insertIfNew! [TransOrd α] (h : t.WF) {k v} :
+    (t.insertIfNew! k v).minKey? =
+      t.minKey?.elim k fun k' => if compare k k' = .lt then k else k' := by
+  simpa only [insertIfNew_eq_insertIfNew!] using minKey?_insertIfNew h
+
+theorem isSome_minKey?_insertIfNew [TransOrd α] (h : t.WF) {k v} :
+    (t.insertIfNew k v h.balanced).impl.minKey?.isSome := by
+  simp_to_model [insertIfNew] using List.isSome_minKey?_insertEntryIfNew
+
+theorem isSome_minKey?_insertIfNew! [TransOrd α] (h : t.WF) {k v} :
+    (t.insertIfNew! k v).minKey?.isSome := by
+  simpa only [insertIfNew_eq_insertIfNew!] using isSome_minKey?_insertIfNew h
+
+theorem minKey?_insertIfNew_le_minKey? [TransOrd α] (h : t.WF) {k v km kmi} :
+    (hkm : t.minKey? = some km) →
+    (hkmi : (t.insertIfNew k v h.balanced |>.impl.minKey? |>.get <| isSome_minKey?_insertIfNew h) = kmi) →
+    compare kmi km |>.isLE := by
+  simp_to_model [insertIfNew] using List.minKey?_insertEntryIfNew_le_minKey?
+
+theorem minKey?_insertIfNew!_le_minKey? [TransOrd α] (h : t.WF) {k v km kmi} :
+    (hkm : t.minKey? = some km) →
+    (hkmi : (t.insertIfNew! k v |>.minKey? |>.get <| isSome_minKey?_insertIfNew! h) = kmi) →
+    compare kmi km |>.isLE := by
+  simpa only [insertIfNew_eq_insertIfNew!] using minKey?_insertIfNew_le_minKey? h
+
+theorem minKey?_insertIfNew_le_self [TransOrd α] (h : t.WF) {k v kmi} :
+    (hkmi : (t.insertIfNew k v h.balanced |>.impl.minKey?.get <| isSome_minKey?_insertIfNew h) = kmi) →
+    compare kmi k |>.isLE := by
+  simp_to_model [insertIfNew] using List.minKey?_insertEntryIfNew_le_self
+
+theorem minKey?_insertIfNew!_le_self [TransOrd α] (h : t.WF) {k v kmi} :
+    (hkmi : (t.insertIfNew! k v |>.minKey?.get <| isSome_minKey?_insertIfNew! h) = kmi) →
+    compare kmi k |>.isLE := by
+  simpa only [insertIfNew_eq_insertIfNew!] using minKey?_insertIfNew_le_self h
+
+theorem minKey?_modify [TransOrd α] [LawfulEqOrd α] {k f} (h : t.WF) :
+    (t.modify k f).minKey? = t.minKey? := by
+  simp_to_model [modify] using List.minKey?_modifyKey
+
+theorem minKey?_alter_eq_self [TransOrd α] [LawfulEqOrd α] (h : t.WF) {k f} :
```

Comment:
Are you planning to add more lemmas for the interaction of `minKey?` and `alter`? _Technically_ the interaction is fully specified by the existing `alter` lemmas and `minKey?_eq_some_iff_mem_and_forall` so it's not strictly necessary, but it might be possible to say something more direct.

Reply:
Hmm, I think adding these lemmas would not make it particularly more ergonomic than going through `mem_alter` and `minKey?_eq_some_iff_mem_and_forall`, so let's just stick with the existing lemmas and add additional lemmas based on real-world examples we encounter.